### PR TITLE
Use ibtypes tests for action key test

### PR DESCRIPTION
**Pull request by matsl**
## What

- Used ibtypes test with action-key instead to check that proper ibtype is used.
- Moved hui tests to own file
- Some doc string updates

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -1,3 +1,25 @@
+2021-04-18  Mats Lidell  <matsl@gnu.org>
+
+* test/hbut-tests.el (hub-mail-address-test, hub-pathname-test)
```
Comment (rswgnu) [inline Changes:3]: What does 'hub' stand for or mean?

Nice overall expansion in test coverage!

File: test/demo-tests.el [hunk 2]
```diff
@@ -238,7 +238,8 @@
 
 ;; org
 (ert-deftest demo-org-hide-header-test ()
-  "Should work with action-key but does not. Works with org modes hide function."
+  "Hide org mode header.
+Bug: Should work with action-key but does not.  Works with org modes hide function."
```
Comment (rswgnu) [inline test/demo-tests.el:242]: Add apostrophe: 'org mode's hide function'.

File: test/hbut-tests.el [hunk 3]
```diff
@@ -165,35 +165,254 @@ the button text"
...
+
+(ert-deftest hub-pathname-emacs-lisp-file-test ()
+  (unwind-protect
+      (with-temp-buffer
+        (insert "\"hyperbole.el\"")
+        (goto-char 2)
+        (action-key)
+        (should (equal major-mode 'emacs-lisp-mode))
+        (should (string= "hyperbole.el" (buffer-name)))))
+  (kill-buffer "hyperbole.el"))
```
Comment (rswgnu) [inline test/hbut-tests.el:223]: Here, should check that buffer-file-name is non-nil to ensure it found the file and didn't just create a buffer with that name.

File: test/hbut-tests.el [hunk 4]
```diff
@@ -165,35 +165,254 @@ the button text"
...
+        (goto-char 2)
+        (action-key)
+        (should (string= "DEMO" (buffer-name)))
+        (should (= (line-number-at-pos) 3))
+        (should (= (current-column) 45)))
+    (kill-buffer "DEMO")))
+
+(ert-deftest hub-pathname-load-path-line-column-test ()
+  "Pathname with line and position specification."
+  (unwind-protect
```
Comment (rswgnu) [inline test/hbut-tests.el:264]: Differentiate doc string from that of prior test.

**Review by rswgnu — CHANGES_REQUESTED** [review]

File: Changes (hunk 1 above)
Comment (matsl) [reply Changes:3]: :-) Massive typo! Will change to `hbut`

File: test/hbut-tests.el [hunk 5]
```diff
@@ -273,13 +273,13 @@ Bug: With line spec looks in the wrong folder for the file?"
     (kill-buffer "hypb.el")))
 
 (ert-deftest hbut-pathname-with-dash-loads-file-test ()
-  "Pathname with dash loads file.
-Bug: Fails with 'Invalid function: hact'."
-  :expected-result :failed
+  "Pathname with dash loads file."
```
Comment (matsl) [inline test/hbut-tests.el:276]: This case fails after Emacs does native byte compilation, I think!? It then complain about "Invalid function: hact'". So changing this to reflect the proper behaviour and leave the byte compilation issue to elsewhere. Running "make clean && make bin && make test-all" seems like a good way to run the tests not to be affected by native compilation.

Comment (matsl) [reply test/hbut-tests.el:276]: Oh, seems still to be a problem with make-all but `"make clean && make bin && make test` seems to be stable anyway.

Must experiment more with turning off the lazy native compilation maybe.

Comment (rswgnu) [reply test/hbut-tests.el:276]: hact is a macro; if it is not loaded before compilation, it is assumed to be a function but then when executed is not defined as a function, hence the error.  So we have to find where it is used but not required and either require it or do so only when compiling.

Comment (matsl) [reply test/hbut-tests.el:276]: As we said in IRL the test case works with not byte compiled hyperbole and with byte compiled hyperbole (after `make bin`). When Emacs asynchronously compiles the byte code to native code the test case fails with the given error message,

Comment (rswgnu) [reply test/hbut-tests.el:276]: Okay, just log to a bug list and we'll close this for now.

**Review by rswgnu — APPROVED** [review]

**Pull request merged**